components/ledger/ledger-persistence/src/integrationTest/kotlin/net/corda/ledger/persistence/consensual/tests/ConsensualLedgerMessageProcessorTests.kt

```diff
@@ -139,25 +143,27 @@ class ConsensualLedgerMessageProcessorTests {
             EXTERNAL_EVENT_CONTEXT.apply {
                 this.contextProperties = keyValuePairListOf(
                     this.contextProperties.toMap() +
-                    cpkFileHashes.toKeyValuePairList(CPK_FILE_CHECKSUM).toMap()
+                            cpkFileHashes.toKeyValuePairList(CPK_FILE_CHECKSUM).toMap()
                 )
             }
         )
 
         // Send request to message processor
-        val processor = LedgerPersistenceRequestProcessor(
+        val processor = LedgerPersistenceRpcRequestProcessor(
             currentSandboxGroupContext,
             virtualNode.entitySandboxService,
             delegatedRequestHandlerSelector,
-            responseFactory
+            responseFactory,
+            requestClass,
+            responseClass
         )
 
         val requestId = UUID.randomUUID().toString()
-        val records = listOf(Record(TOPIC, requestId, request))
+        listOf(Record(TOPIC, requestId, request))
 
         // Process the messages (this should persist transaction to the DB)
-        var responses = assertSuccessResponses(processor.onNext(records))
-        assertThat(responses).hasSize(1)
+        var response = assertSuccessResponse(processor.process(request), logger)
+        assertThat(response).isNotNull
```

Comment:
```suggestion
        assertSuccessResponse(processor.process(request), logger)
```
Nit: The return type of `LedgerPersistenceRpcRequestProcessor::process` isn't nullable, so you don't need this check. The `::assertSuccessResponse` function you added should do it